### ISSUE TITLE
[HOLD for payment 2024-04-09] [$500] [Held requests] show status on distance requests

### ISSUE THREAD

**Issue by robertjchen** [Expensify/App#37476]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Coming from the discussion here: https://github.com/Expensify/App/pull/37121#issuecomment-1965819951

Update display of  status on distance requests to show state (Approved/Pending/Canceled/Hold) like we do for Cash expenses. 

**Version Number:** 
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1710439388757009

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace chat
3. Create a distance request
4. Go to request details page
5. Click 3-dot menu > Hold request
6. Add a reason and save it

## Expected Result:
Distance expense request should show `Hold` 

## Actual Result:
Distance expense request does not show any status 

## Workaround:
N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

![image](https://github.com/Expensify/App/assets/2438855/7a363092-2f3a-42fa-a185-11a521f0f0d1)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e3e32c45e45f949f</li>
        <li>Upwork Job ID: 1769963579385311232</li>
        <li>Last Price Increase: 2024-03-19</li>
<li>Automatic offers: </li>
<ul>
<li>situchan | Reviewer | 0</li>
<li>ShridharGoel | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by stephanieelliott** [comment]
Hey @robertjchen just want to check where we landed with this one from the original discussion in https://github.com/Expensify/App/pull/37121#issuecomment-1965819951 so we can figure out the `Actual Result` vs `Intended Result`

> @robertjchen for now, • xxx is showing only after Cash (manual request)
i.e. not showing for distance requests. Is this expected (or at least out of scope)?

> > @situchan great catch! Let's pull this discussion to Slack to determine what's the best course of action. I think this change can go out in the interim however 👍


Was this ever pulled into Slack (I can't find anything)? Or was there general agreement that distance expenses should show the status (Hold, Approved, Pending, Canceled etc) on  distance expenses?

**Comment by robertjchen** [comment]
thanks @situchan ! Feel free to link the thread here after you've created it, looking forward to see what others think

**Comment by stephanieelliott** [comment]
Hey @situchan have you started this convo? I'm not seeing it in Slack -- could you link it here?

**Comment by stephanieelliott** [comment]
@situchan has this convo been started? Is there anything we can do to move this issue along in the meantime?

**Comment by situchan** [comment]
started discussion - https://expensify.slack.com/archives/C01GTK53T8Q/p1710439388757009

**Comment by situchan** [comment]
We got answer. It should be consistent. Let's fix this to show status on all type of requests.

**Comment by situchan** [comment]
Not overdue

**Comment by stephanieelliott** [comment]
Hey @situchan  we have a couple proposals here -- can you review when you get a chance?

**Comment by situchan** [comment]
No one considered all cases in their proposed solutions.
I don't understand why other types of requests (i.e. scan) cannot have status labels.

i.e. request while scanning in progress can be held.

https://github.com/Expensify/App/assets/108292595/8323ed20-adb5-4075-8813-fa41d3327cb2

I'm looking for proposal which filled all of these:
|  | Approved | Pending | Canceled | Hold | Review required | Missing amount | Missing merchant
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| Manual  |
| Distance  |
| Scanning  |
| Scan  |
| Split  |
| Card  |

**Comment by situchan** [comment]
@ShridharGoel are you sure that all labels are applicable to all types of requests? Did you test? 😄

**Comment by situchan** [comment]
> I can still update update my proposal right? @situchan

I think @ShridharGoel is the first dib unless you have any other meaningful considerations.

**Comment by situchan** [comment]
Sounds good. Thanks

**Comment by situchan** [comment]
@ShridharGoel I also expect ✔️ in all fields. Just wanted to make sure that the solution doesn't cause any regressions.
As promised, let's wait for @GandalfGwaihir a few hrs. Otherwise, we can go with your proposal.

**Comment by situchan** [comment]
@rayane-djouah it was discussed already. If you have any doubt, please ask in slack thread.

**Comment by situchan** [comment]
@ShridharGoel's [proposal](https://github.com/Expensify/App/issues/37476#issuecomment-2005974421) looks good to me.
🎀 👀 🎀 C+ reviewed

**Comment by robertjchen** [comment]
Thanks for the discussion here and on Slack, and for the updated detailed proposal. Let's move forward with this 👍

**Comment by stephanieelliott** [comment]
PR is undergoing QA now

**Comment by trjExpensify** [comment]
Hit staging 4 days ago. Should start the regression clock on the next deploy.

**Comment by stephanieelliott** [comment]
Summarizing payment on this issue: 

 - Contributor: @ShridharGoel please accept the Upwork offer so we can issue payment 
 - Contributor+: @situchan $500 via Upwork PAID 
 
 Upwork job is here: https://www.upwork.com/jobs/~01e3e32c45e45f949f

**Comment by stephanieelliott** [comment]
All paid! See summary here: https://github.com/Expensify/App/issues/37476#issuecomment-2045899200
